libraries/python/chat-driver/chat_driver/openai_chat_completion_driver.py

```diff
@@ -59,16 +105,10 @@ def __init__(self, config: ChatDriverConfig) -> None:
         # Set up logging.
         self.logger = Logger(__name__, self.context)
 
-        # Load messages from file if they exist.
+        # Load messages from message provider if they exist.
         init_messages: list[ChatCompletionMessageParam] = config.messages or []
-        self.messages_file = self.session_dir / "messages.json"
-        if not self.messages_file.exists():
-            self.messages_file.write_text(json.dumps(init_messages, indent=2))
-            self.messages = init_messages
-        else:
-            messages = json.loads(self.messages_file.read_text())
-            messages.extend(init_messages)
-            self.messages = messages
+        self.messages_provider = config.message_provider or LocalMessageHistoryProvider(self.session_dir)
+        self.messages_provider.extend(init_messages)
```

Comment:
it seems really weird to provide messages on the ChatDriverConfig, and a messagehistoryprovider, and the chatdriver just then adds them to the message history.

what is the intent of the "messages" on the ChatDriverConfig? i wonder if it is needed at all?

Reply:
Good call. Moving to the message provider.